contract/migrations/12.0.2.0.0/pre-migration.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import SUPERUSER_ID, api
+
+_logger = logging.getLogger(__name__)
+
+
+def migrate(cr, version):
+    """
+    set recurring_next_date to false for finished contract
+    """
+    _logger.info("order all contract line")
+    with api.Environment(cr, SUPERUSER_ID, {}) as env:
```

Comment:
I am getting an error when running the migration here: 

AttributeError: __exit__

But I am not sure what is wrong

Reply:
Hi @aheficent ,
I pushed a new version of the migration scripts. I think it resolve this issue